akka-stream/src/main/scala/akka/stream/Attributes.scala

```diff
@@ -439,6 +439,46 @@ object Attributes {
       strategy: CancellationStrategy.Strategy): CancellationStrategy.Strategy =
     CancellationStrategy.AfterDelay(delay, strategy)
 
+  /**
+   * Nested materialization cancellation strategy provides a way to configure the cancellation behavior of stages that materialize a nested flow at some stage.
+   *
+   * When cancelled before materializing their nested flows, these stages can either immediately cancel (default behaviour) without materializing the nested flow
+   * or wait for the nested materialization and then propagate the cancellation signal.
+   *
+   * This applies to [[akka.stream.scaladsl.FlowOps.flatMapPrefix]], [[akka.stream.scaladsl.Flow.futureFlow]] (and derivations such as [[akka.stream.scaladsl.Flow.lazyFutureFlow]]).
+   * These operators either delay the nested flow's materialization or wait for a future to complete before doing so,
+   * in this period of time they may receive a downstream cancellation signal. When this happens these operators will behave according to
+   * this [[Attribute]]: when set to true they will 'stash' the signal and later deliver it to the materialized nested flow
+   * , otherwise these stages will immediately cancel without materializing the nested flow.
+   */
+  @ApiMayChange
+  case class NestedMaterializationCancellationPolicy(propagateToNestedMaterialization: Boolean)
```

Comment:
API-wise, maybe we should make the constructor private[akka] here to hide the internals? And then have constants in `NestedMaterializationCancellationPolicy`.

 * `NestedMaterializationCancellationPolicy.EagerCancellation`
 * `NestedMaterializationCancellationPolicy.PropagateToNested`

WDYT?

Reply:
Same for Java API.

Reply:
so Java API will look like: `NestedMaterializationCancellationPolicyPropagateToNested` ?
not vey welcoming, is it?

Reply:
@jrudolph can u please reply to this?

Reply:
I don't have a better suggestion, let's better be explicit.